### ISSUE TITLE
Installation error

### ISSUE THREAD

**Issue by maartenvant** [CONNCTED/homebridge-soundtouch#18]
Hi,

I tried to install it, I installed already several plugins, but with the Bose I get the following error:

Anybody know how to solve?

gyp ERR! configure error 
gyp ERR! stack Error: EACCES: permission denied, mkdir '/usr/local/lib/node_modules/homebridge-soundtouch/node_modules/mdns/build'
gyp ERR! System Darwin 17.3.0
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /usr/local/lib/node_modules/homebridge-soundtouch/node_modules/mdns
gyp ERR! node -v v8.9.4
gyp ERR! node-gyp -v v3.6.2
gyp ERR! not ok 
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! mdns@2.3.4 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the mdns@2.3.4 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

**Comment by Seger85** [comment]
+ 1

I have the same Issues:

https://github.com/CONNCTED/homebridge-soundtouch/issues/17

#17

Seger

**Comment by cameronfergus** [comment]
SAME ISSUE!!! Something to do with the mdns module I'm thinking... Tried to install it separately and had similar issues. Can you confirm? Thanks!

**Comment by Seger85** [comment]
"Reinstall Npm with a Version Manager (Recommended)" 

Hey, can not you help me with this? how do I install the recommended version under Synology, do I simply go through a package manager and install the "version manager" or do I go over the 2 variant? 

I hope not, as I do not feel comfortable on the console at all. If you have to go over the 2 varinates, can you maybe do it slowly?

Thanks for your support! 

Best regards Seger

**Comment by cameronf87** [comment]
I followed the second option (with the console). I am not exactly sure what to do for your DS however. I have a Synology also and I have used the terminal a few times for various settings, etc. I know you can sudo and there aren't restrictions, so I would think you could run all of these commands without issues. 

I would just copy/paste into your terminal window and try these steps first. It seems to have fixed other issues I was having trying to install other NPM packages, so this was a good fix. 

It basically moves the npm "global" location from it's path outside of your home folder (somewhere like /usr/local/bin) and puts it in your home folder (~/), so it doesn't require sudo at all for downloading packages. It makes sense why this is better, but not sure how it's still "global" if it's sitting in your home folder, but in any case it does seem to work.

**Comment by Seger85** [comment]
I have done it after the 2 variant but in the 2 position comes immediately an error that he does not know the command. I was logged in under root and on my admin user, but both times the error comes.

Furthermore, I have the phenomenon that if I let out the plugin from my config and install file, that it is trozdem in the terminal to the error:


![image](https://user-images.githubusercontent.com/35084198/36218497-d42e2c44-11b4-11e8-81b3-9747b56df1e2.png)



As soon as I add the plugin again normally, the known bug comes from the #17 #18

Maybe someone can help me, I've tried everything. Gladly also via Teamviewer, if it brings something. I speak only German or very bad English.

Thanks and best regards
Seger

**Comment by cameronf87** [comment]
Try the 2nd option again, only this time don't login as root, or use the "sudo su" command, this is meant to be installed by your user account (admin). Important note: in Unix/Linux, the "root" user is different than "admin" or a superuser. The "root" user should be used with caution as you can remove system files, and do a lot of things with unintended consequences. 

You should be able to run these commands while logged in as the "admin" user and using the sudo command to execute the command as root (different than logging in as root), this "should" execute the command as root, but not cause permission errors. I'm not an expert with permissions and specifically your flavor of Linux that runs on the Synology.

 If that fails, try installing the package without the global (-g) flag. To do this, first make sure you uninstall the current package, so you would do "sudo npm uninstall -g homebridge-soundtouch". Then you would do "npm install homebridge-soundtouch" if you get a permission error, try "sudo npm install homebridge-soundtouch". If all of those solutions fail, you may need to reinstall npm completely under your home folder, and not as the root user. 

See this article for instruction on how to do this. https://increaseyourgeek.wordpress.com/2010/08/18/install-node-js-without-using-sudo/

Long story short: You have permission issues with either npm itself, or the location it's installed, if you keep it all under your home folder (~/) then you shouldn't have any issues.
